### ISSUE TITLE
Kernel Panic - Centos 7.2 3.10.0 - with devicemapper LVM thinpool ext4

### ISSUE THREAD

**Issue by sakserv** [docker/docker#25504]
**Output of `docker version`:**

```
Client:
 Version:      1.11.2
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   b9f10c9
 Built:        Wed Jun  1 21:23:11 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.11.2
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   b9f10c9
 Built:        Wed Jun  1 21:23:11 2016
 OS/Arch:      linux/amd64
```

**Output of `docker info`:**

```
Containers: 178
 Running: 5
 Paused: 0
 Stopped: 173
Images: 8
Server Version: 1.11.2
Storage Driver: devicemapper
 Pool Name: vg01-docker--pool
 Pool Blocksize: 524.3 kB
 Base Device Size: 274.9 GB
 Backing Filesystem: ext4
 Data file:
 Metadata file:
 Data Space Used: 1.655 TB
 Data Space Total: 5.63 TB
 Data Space Available: 3.976 TB
 Metadata Space Used: 119.7 MB
 Metadata Space Total: 16.98 GB
 Metadata Space Available: 16.86 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Library Version: 1.02.107-RHEL7 (2015-12-01)
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge null host
Kernel Version: 3.10.0-327.13.1.el7.x86_64
Operating System: CentOS Linux 7 (Core)
OSType: linux
Architecture: x86_64
CPUs: 32
Total Memory: 251.6 GiB
Name: XXXXXXXXXXX
ID: AISM:7SDV:XXJ2:Z22V:LPV7:VQNL:XOWS:6J65:WRGX:SB2H:P4YT:VL7K
Docker Root Dir: /grid/0/docker
Debug mode (client): false
Debug mode (server): false
Username: XXXXXX
Registry: https://index.docker.io/v1/
WARNING: bridge-nf-call-iptables is disabled
WARNING: bridge-nf-call-ip6tables is disabled
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
CentOS Linux release 7.2.1511 (Core)
3.10.0-327.13.1.el7.x86_64
Physical

**Issue:**
We are encountering sporadic kernel panics on roughly one node per day out of ~100. The oops output is as follows:

```
[363954.342003] kernel BUG at fs/buffer.c:1270!
[363954.342238] invalid opcode: 0000 [#1] SMP 
[363954.342487] Modules linked in: ts_bm xt_string tcp_diag inet_diag binfmt_misc veth xt_addrtype dm_thin_pool dm_persistent_data dm_bio_prison dm_bufio libcrc32c loop ip6t_rpfilter ip6t_REJECT ipt_REJECT xt_conntrack ebtable_nat ebtable_broute bridge stp llc ebtable_filter ebtables ip6table_nat nf_conntrack_ipv6 nf_defrag_ipv6 nf_nat_ipv6 ip6table_mangle ip6table_security ip6table_raw ip6table_filter ip6_tables iptable_nat nf_conntrack_ipv4 nf_defrag_ipv4 nf_nat_ipv4 nf_nat nf_conntrack iptable_mangle iptable_security iptable_raw iptable_filter ip_tables ipmi_devintf iTCO_wdt iTCO_vendor_support dcdbas intel_powerclamp coretemp kvm_intel kvm crct10dif_pclmul crc32_pclmul crc32c_intel ghash_clmulni_intel aesni_intel lrw gf128mul glue_helper ablk_helper cryptd pcspkr sb_edac edac_core ipmi_si ipmi_msghandler
[363954.344968]  acpi_pad wmi acpi_power_meter mei_me lpc_ich mei mfd_core shpchp ext4 mbcache jbd2 sd_mod sr_mod crc_t10dif cdrom crct10dif_common mgag200 syscopyarea sysfillrect sysimgblt i2c_algo_bit drm_kms_helper ttm ahci drm libahci libata i2c_core tg3 megaraid_sas ptp pps_core dm_mirror dm_region_hash dm_log dm_mod
[363954.346682] CPU: 10 PID: 9848 Comm: wgctl Tainted: G        W   --------------   3.10.0-229.el7.x86_64 #1
[363954.347297] Hardware name: Dell Inc. PowerEdge R720/0DCWD1, BIOS 2.2.2 01/16/2014
[363954.347933] task: ffff880fe3072220 ti: ffff880d8b068000 task.ti: ffff880d8b068000
[363954.348609] RIP: 0010:[<ffffffff81601fc6>]  [<ffffffff81601fc6>] check_irqs_on.part.11+0x4/0x6
[363954.349329] RSP: 0018:ffff880d8b06b7e8  EFLAGS: 00010046
[363954.350025] RAX: 0000000000000096 RBX: 00000000000063c7 RCX: 00000000009950cb
[363954.350747] RDX: 0000000000001000 RSI: 00000000009950cb RDI: ffff8807fe3a0000
[363954.351478] RBP: ffff880d8b06b7e8 R08: 0000000000000000 R09: 0000000000000000
[363954.352221] R10: 0000000000000000 R11: ffff880720bb4300 R12: ffff8807fe3a0000
[363954.352978] R13: 0000000000001000 R14: 0000000000000000 R15: 0000000000000000
[363954.353747] FS:  00007f0d24e60700(0000) GS:ffff88080faa0000(0000) knlGS:0000000000000000
[363954.354537] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[363954.355339] CR2: 00007fab9c4f6fb8 CR3: 000000000190a000 CR4: 00000000001407e0
[363954.356164] DR0: 0000000000000000 DR1: 0000000000000000 DR2: 0000000000000000
[363954.356999] DR3: 0000000000000000 DR6: 00000000ffff0ff0 DR7: 0000000000000400
[363954.357835] Stack:
[363954.358672]  ffff880d8b06b808 ffffffff811fa3a5 00000000000063c7 ffff8807fe3a0000
[363954.359560]  ffff880d8b06b868 ffffffff811fa3d5 00000000000526fe 000000002fd2f61b
[363954.360497]  ffff88071c482338 ffff88071c482350 00000000009950cb 00000000000063c7
[363954.361403] Call Trace:
[363954.362352]  [<ffffffff811fa3a5>] __find_get_block+0x115/0x120
[363954.363286]  [<ffffffff811fa3d5>] __getblk+0x25/0x2e0
[363954.364245]  [<ffffffffa025f7c1>] __read_extent_tree_block+0x41/0x200 [ext4]
[363954.365212]  [<ffffffff811acb53>] ? __kmalloc+0x1f3/0x230
[363954.366195]  [<ffffffffa0260e3c>] ext4_ext_find_extent+0x12c/0x2e0 [ext4]
[363954.367208]  [<ffffffffa026450e>] ext4_ext_map_blocks+0x7e/0x1220 [ext4]
[363954.368214]  [<ffffffff8116547f>] ? __pagevec_release+0x2f/0x40
[363954.369246]  [<ffffffffa0235f72>] ? mpage_prepare_extent_to_map+0x2b2/0x2e0 [ext4]
[363954.370294]  [<ffffffffa0236d6d>] ext4_map_blocks+0x16d/0x560 [ext4]
[363954.371351]  [<ffffffffa023a272>] ext4_writepages+0x692/0xd60 [ext4]
[363954.372410]  [<ffffffff81162f5e>] do_writepages+0x1e/0x40
[363954.373515]  [<ffffffff81157db5>] __filemap_fdatawrite_range+0x65/0x80
[363954.374622]  [<ffffffff81157e7c>] filemap_flush+0x1c/0x20
[363954.375737]  [<ffffffffa0237adc>] ext4_alloc_da_blocks+0x2c/0x70 [ext4]
[363954.376855]  [<ffffffffa0230f99>] ext4_release_file+0x79/0xc0 [ext4]
[363954.377978]  [<ffffffff811c8849>] __fput+0xe9/0x270
[363954.379089]  [<ffffffff811c8b1e>] ____fput+0xe/0x10
[363954.380196]  [<ffffffff81093c67>] task_work_run+0xa7/0xe0
[363954.381351]  [<ffffffff8107467b>] do_exit+0x2cb/0xa60
[363954.382431]  [<ffffffff81081c23>] ? __sigqueue_free.part.11+0x33/0x40
[363954.383488]  [<ffffffff8108221c>] ? __dequeue_signal+0x13c/0x220
[363954.384515]  [<ffffffff81074e8f>] do_group_exit+0x3f/0xa0
[363954.385511]  [<ffffffff810851e0>] get_signal_to_deliver+0x1d0/0x6e0
[363954.386484]  [<ffffffff81013467>] do_signal+0x57/0x6c0
[363954.387438]  [<ffffffff81013b39>] do_notify_resume+0x69/0xb0
[363954.388391]  [<ffffffff8160c1fc>] retint_signal+0x48/0x8c
[363954.389270] Code: c7 c7 a8 f4 83 81 e8 76 c5 ff ff 4d 85 e4 74 12 49 8d 7c 24 58 e8 8b 9c 00 00 4c 89 e7 e8 33 b7 bd ff 5b 41 5c 5d c3 55 48 89 e5 <0f> 0b 55 48 89 e5 0f 0b 0f 1f 44 00 00 55 48 89 e5 0f 0b 55 48 
[363954.391222] RIP  [<ffffffff81601fc6>] check_irqs_on.part.11+0x4/0x6
[363954.392135]  RSP <ffff880d8b06b7e8>
```

We are leveraging devicemapper with a backing LVM thinpool using the ext4 filesystem.

```
  LV          VG   Attr       LSize Pool Origin Data%  Meta%  Move Log Cpy%Sync Convert
  docker-pool vg01 twi-aot--- 5.12t             29.40  0.71
```

Our workload is similar to a Jenkins workload, QE testing, but the containers are not cleaned up on exit to allow for additional debugging. As found in the output, we have about 175 stopped containers on this node that recently panic'd, in case this might be a factor.

In looking at the changelog for the latest available 3.10 kernel on Centos 7.2, I don't see any fixes that appear related, so I'm hesitant to just roll out a kernel upgrade in hopes it fixes it. I've been unable to locate similar panics in my searching.

Has anyone encountered this or a similar panic before and have thoughts on root cause/next steps? Happy to provide any additional details as requested.

**Comment by thaJeztah** [comment]
I haven't seen this reported, but given that docker doesn't install kernel modules, this is unlikely a bug in the docker codebase.

ping @rhvgoyal anything familiar in this?

**Comment by sidharta-s** [comment]
possibly related to this? https://lkml.org/lkml/2013/10/31/385

**Comment by rhvgoyal** [comment]
Can you try latest centos kernel and see if this issue is already fixed or not.

**Comment by sakserv** [comment]
We ended up moving to xfs for the filesystem and no additional panics have occurred. 

However, a colleague reached out to some kernel devs he knows and the consensus is that the panics are related to the LSI megaraid driver. Several panics have been fixed in versions newer than what we are running, including [SCGCQ00827403](https://delivery04.dhe.ibm.com/sar/CMA/XSA/0632l/1/lnvgy_dd_sraidmr_6.808.16.00_sles12_32-64.chg), which very much sounds like the panic we are seeing.

We can close this for now, but hopefully the above will prove useful for others.
